odoo/tools/mail.py

```diff
@@ -177,19 +105,125 @@ def parse_style(self, el):
+                node.insert(node_idx, new_node)
+            else:
+                child_node.tail = text[idx:item.start()]
+                new_node.tail = text[item.end():]
+                node.insert(node_idx, new_node)
+            child_node = new_node
+            idx = item.end()
+            node_idx = node_idx + 1
+
+    el_class = el.get('class', '') or ''
+    el_id = el.get('id', '') or ''
+
+    # gmail or yahoo // # outlook, html // # msoffice
+    if 'gmail_extra' in el_class or \
+            'divRplyFwdMsg' in el_id or \
+            ('SkyDrivePlaceholder' in el_class or 'SkyDrivePlaceholder' in el_class):
+        el.set('data-o-mail-quote', '1')
+        if el.getparent() is not None:
+            el.getparent().set('data-o-mail-quote-container', '1')
+
+    if (el.tag == 'hr' and ('stopSpelling' in el_class or 'stopSpelling' in el_id)) or \
+       'yahoo_quoted' in el_class:
+        # Quote all elements after this one
+        el.set('data-o-mail-quote', '1')
+        for sibling in el.itersiblings(preceding=False):
+            sibling.set('data-o-mail-quote', '1')
+
+    # html signature (-- <br />blah)
+    signature_begin = re.compile(r"((?:(?:^|\n)[-]{2}[\s]?$))")
+    if el.text and el.find('br') is not None and re.search(signature_begin, el.text):
+        el.set('data-o-mail-quote', '1')
+        if el.getparent() is not None:
+            el.getparent().set('data-o-mail-quote-container', '1')
+
+    # text-based quotes (>, >>) and signatures (-- Signature)
+    text_complete_regex = re.compile(r"((?:\n[>]+[^\n\r]*)+|(?:(?:^|\n)[-]{2}[\s]?[\r\n]{1,2}[\s\S]+))")
+    if not el.get('data-o-mail-quote'):
+        _tag_matching_regex_in_text(text_complete_regex, el, 'span', {'data-o-mail-quote': '1'})
+
+    if el.tag == 'blockquote':
+        # remove single node
+        el.set('data-o-mail-quote-node', '1')
+        el.set('data-o-mail-quote', '1')
+    if el.getparent() is not None and (el.getparent().get('data-o-mail-quote') or el.getparent().get('data-o-mail-quote-container')) and not el.getparent().get('data-o-mail-quote-node'):
+        el.set('data-o-mail-quote', '1')
+
+
+def html_normalize(src, pre_sanitize=True, post_sanitize=True, silent=True):
+    if not src:
+        return src
+
+    logger = logging.getLogger(__name__ + '.html_normalize')
+
+    if pre_sanitize:
+        src = ustr(src, errors='replace')
+        # html: remove encoding attribute inside tags
+        doctype = re.compile(r'(<[^>]*\s)(encoding=(["\'][^"\']*?["\']|[^\s\n\r>]+)(\s[^>]*|/)?>)', re.IGNORECASE | re.DOTALL)
+        src = doctype.sub(u"", src)
+
+        try:
+            # perform quote detection before cleaning and class removal
+            doc = html.fromstring(src)
+            if doc is not None:
+                for el in doc.iter(tag=etree.Element):
+                    tag_quote(el)
+        # TODO: Those `except` cases are copy/paster from `html_sanitize`, fix that
```

Comment:
Waiting for first review before tackling this one (probably going to use `@wraps` here?).
(I am not sure the "empty" case is actually still needed in the other place.)
Probably someone will have something to say about `def html_normalize(src, pre_sanitize=True, post_sanitize=True, silent=True):` implementation, following that I'll adapt.

Reply:
Ok, I misunderstood what you meant by 

> But fundamentally the entire silent feature seems unnecessary to maintain?

Adapted the way you meant it.